Sources/Extensions/Matter/RequestHandler.swift

```diff
@@ -0,0 +1,63 @@
+import HAKit
+import MatterSupport
+import PromiseKit
+import Shared
+
+// The extension is launched in response to `MatterAddDeviceRequest.perform()` and this class is the entry point
+// for the extension operations.
+class RequestHandler: MatterAddDeviceExtensionRequestHandler {
+    override func validateDeviceCredential(
+        _ deviceCredential: MatterAddDeviceExtensionRequestHandler.DeviceCredential
+    ) async throws {
+        // Use this function to perform additional attestation checks if that is useful for your ecosystem.
+    }
+
+    override func selectWiFiNetwork(from wifiScanResults: [
+        MatterAddDeviceExtensionRequestHandler.WiFiScanResult
+    ]) async throws -> MatterAddDeviceExtensionRequestHandler.WiFiNetworkAssociation {
+        .defaultSystemNetwork
+    }
+
+    override func selectThreadNetwork(from threadScanResults: [
+        MatterAddDeviceExtensionRequestHandler.ThreadScanResult
+    ]) async throws -> MatterAddDeviceExtensionRequestHandler.ThreadNetworkAssociation {
+        .defaultSystemNetwork
+    }
+
+    override func commissionDevice(
+        in home: MatterAddDeviceRequest.Home?,
+        onboardingPayload: String,
+        commissioningID: UUID
+    ) async throws {
+        guard let payload = Int(onboardingPayload) else {
+            struct SomeError: Error {}
+            throw SomeError()
+        }
+
+        try await withCheckedThrowingContinuation { continuation in
+            when(resolved: Current.apis.map { api in
+                api.connection.send(.matterComissionOnNetwork(pin: payload)).promise.map { _ in () }
```

Comment:
this should be a "commission with code" where you send in the code (string) you retrieved from the payload.